parallel_algos/memory_tracking_allocator.cuh

```diff
@@ -55,9 +62,10 @@ public:
         num_allocs.store(0);
     }
 
+    template <bool IsTemporary>
     void print_stats() const
     {
-        std::printf("Memory statistics:\n");
+        std::printf("%s memory statistics:\n", IsTemporary ? "Temporary" : "Input/Output");
```

Comment:
`IsTemporary` could instead be passed as a label

```suggestion
void print_stats(std::string_view name) const
{
        std::printf("%s memory statistics:\n", name);
```
?

Or even stored as member of the memory tracker?